File: web_service/exercicio_01/Python/exercicio_01.py
#coding: utf-8
import requests
from sys import argv
import json

def requisicao(nome):
    r = requests.get('https://api.github.com/users/' + nome)
    if 200 == r.status_code:
        s = r.content
        s = json.loads(s)
        nome = 'Não Definido'
        email = 'Não Definido'
        seguidores = 'Não Definido'
        if 'name' in s:
            nome = s['name']
        if 'email' in s:
            email = s['email']
        if 'email' in s:
            seguidores = s['following']
        print 'Nome: {0}\nE-mail: {1}\nSeguidores: {2}'.format(nome, email, seguidores)
    else:
        print 'Usuário inexistente'

if __name__ == '__main__':

    if len(argv) == 2:
    	requisicao(argv[1])
    else:
        print 'Parâmetros: python exercicio_01.py nome_usuario'

File: web_service/exercicio_01/Java/exercicio_01.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

import com.google.gson.Gson;

public class exercicio_01 {

	public GitHubUser obterDadosGitHubUser(String nome) throws Exception {
		HttpURLConnection conn;
		BufferedReader rd;
		URL url = new URL("https://api.github.com/users/" + nome);
		conn = (HttpURLConnection) url.openConnection();
		conn.setRequestMethod("GET");
		if (conn.getResponseCode() == 200) {
			rd = new BufferedReader(new InputStreamReader(conn.getInputStream()));
			GitHubUser usuario = new Gson().fromJson(rd, GitHubUser.class);
			rd.close();
			return usuario;
		}
		else
			return null;
	}

	public static void main(String args[]) {
		if (args.length == 1) {
			exercicio_01 exerc = new exercicio_01();
			GitHubUser usuario = null;
			try {
				usuario = exerc.obterDadosGitHubUser(args[0]);
			} catch (Exception e) {
				System.out.println("Ocorreu um erro: " + e.getMessage());
			}
			if (usuario == null)
				System.out.println("Usuário Inexistente");
			else
				System.out.println(usuario.toString());
		}
		else
			System.out.println("Parâmetros: java exercicio_01 nome_usuario");
	}
}

class GitHubUser {
	protected String name;
	protected String email;
	protected int followers;

	public String toString() {
		return "Nome: " + name + "\nE-mail: " + email + "\nSeguidores: " + followers;
	}
}
